rllib/algorithms/algorithm_config.py

```diff
@@ -619,6 +631,23 @@ def update_from_dict(
 
         return self
 
+    def serialize(self) -> Mapping[str, Any]:
```

Comment:
TODO: We might want to have an `deserialize` as well. Right now, simply using the `from_dict()` API works in this same (deserializing) manner, whether the dict used is actually code-free (already serialized) or not (i.e. a classic RLlib config dict with e.g. "callbacks" key still pointing to a class).

We should get rid of user-facing dicts altogether and only use dicts for under-the-hood serializations, e.g. when we need to create msgpack checkpoints.

Reply:
I agree. Can we add this exact todo in the code that we can keep track of this even after the pr is closed. Or an open issue would also help.